demo/admin/src/common/pages/page.type.ts

```diff
@@ -0,0 +1,14 @@
+import { MenuItemRouterLinkProps } from "@comet/admin";
+import { UserPermission } from "@comet/cms-admin";
+import { RouteProps } from "react-router";
+
+export type Page = {
+    requiredPermission?: UserPermission;
+    routes?: (RouteProps & { requiredPermission?: UserPermission })[];
+    route?: RouteProps;
```

Comment:
Couldn't this be merged into `routes?: (RouteProps & { requiredPermission?: UserPermission }) | (RouteProps & { requiredPermission?: UserPermission })[]`? What would happen if I specify both route and routes?

Reply:
I updated the logic and created two custom hooks in cms-admin: https://github.com/vivid-planet/comet/pull/1223/commits/9b45903bc76070e2e4cd8d1c2ee108654286338a

cms-admin now provides:

- Type `Pages`
- Hook `useMenuForCurrentUser`
- Hook `useRoutesForCurrentUser`

I'm not sure about the naming for the typ since `Page` is already used for the content. Do you have suggestions or is it okay like this?